### PR TITLE
SHS-5492: Fix bug in hb-stretch-vertical-linked-cards

### PR DESCRIPTION
**Pull request by mariannuar**
# READY FOR REVIEW

## Summary
Fix `hb-stretch-vertical-linked-cards` class to cause the cards to be the same height and the titles to all be the same height

## Need Review By (Date)
02/28

## Urgency
high

## Steps to Test
1. Go to https://english.suhumsci.loc/
2. Log in and go to https://english.suhumsci.loc/node/2/layout
3. Remove the `.hb-raised-cards--uniform-height` class in the `Vertical Linked Cards`
4. Confirm the cards and their titles have the same height

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss [hunk 1]
```diff
@@ -238,3 +225,11 @@
...
   }
 }
+
+.hb-stretch-vertical-linked-cards {
+  height: 100%;
+
+  .field-hs-collection-items .ptype-hs-postcard {
+    display: flex;
+  }
+}
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: These styles only work for a very specific markup structure. Isn't it possible to have the same effect with a more generic approach? The `hb-stretch-vertical-linked-cards` can be used in other context, such as views.

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Looks good! Just a minor detail to review. Let me know if you have questions.

File: docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss (hunk 1 above)
Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: @cienvaras I can check and add a more generic approach. Do you have an example where that's being used in another context? Just to be sure that what I do, works as expected

Comment (cienvaras) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: @mariannuar  Don't have a specific example right now, but according to the [Utility class documentation](https://github.com/SU-HSDO/suhumsci/blob/develop/docroot/themes/humsci/humsci_basic/docs/utility-classes.md) it's possible to use this class in views. Imagine something like this:

```
<div class="hb-stretch-vertical-linked-cards">
  // Any markup structure in the middle.
  <div class="hb-vertical-linked-card">
    ...
  </div>
</div>
```

Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss:235]: Oh ok. Thanks!

Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: @cienvaras Oh got it. Thanks :)

**Comment by mariannuar** [comment]
@cienvaras I made that piece of code more generic, let me know if that's better. I test it and it still works, and it doesn't seem to be affecting other div. Let me know if you find something. Thanks!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss [hunk 2]
```diff
@@ -229,7 +229,7 @@
 .hb-stretch-vertical-linked-cards {
   height: 100%;
 
-  .field-hs-collection-items .ptype-hs-postcard {
+  & > div > div > div > div {
     display: flex;
   }
 }
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: Sorry, but this doesn't work. You're assuming a specific number of divs, this is probably less generic than the previous implementation.

Why do we need the `display: flex` here? I tried removing it and the english site works well.

**Review by cienvaras — CHANGES_REQUESTED** [review]

File: docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss (hunk 2 above)
Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: @cienvaras I don't know why but at the beginning it wasn't working without that. But now that you mentioned, I removed it and tested, and you're right. Thanks for catching it up!

**Comment by mariannuar** [comment]
@cienvaras Ready for review again (sorry 😣)

File: docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss [hunk 3]
```diff
@@ -238,3 +225,7 @@
     }
   }
 }
+
+.hb-stretch-vertical-linked-cards {
+  height: 100%;
+}
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_vertical-linked-card.scss]: Just one final change before merging. Please remove this too, is not needed and might cause problems on some cases.

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Thanks! Just one final change and we're good to go :)

**Comment by mariannuar** [comment]
@cienvaras Done :)

**Review by cienvaras — APPROVED** [review]
@mariannuar Thanks! :shipit:

**Pull request merged**